lib/transport/import_data_service.ex

```diff
@@ -0,0 +1,331 @@
+
+  ## Examples
+      iex> [%{"mime" => "text/csv"}]
+      ...> |> ImportDataService.filter_csv()
+      [%{"mime" => "text/csv"}]
+
+      iex> [%{"mime" => "text/cv"}]
+      ...> |> ImportDataService.filter_csv()
+      []
+
+      iex> [%{"mime" => "text/csv"}, %{"mime" => "application/neptune"}]
+      ...> |> ImportDataService.filter_csv()
+      [%{"mime" => "text/csv"}]
+
+  """
+  def filter_csv(r) do
+    Enum.filter(r, fn %{"mime" => mime} -> mime == "text/csv"
+                      _                 -> false end)
+  end
+
+  @doc """
+  filter csv http response
+
+  ## Examples
+      iex> {:ok, %{headers: [{"Content-Type", "text/csv"}]}}
+      ...> |> ImportDataService.filter_header_csv
+      true
+
+      iex> {:ok, %{headers: [{"Content-Type", "application/zip"}]}}
+      ...> |> ImportDataService.filter_header_csv
+      false
+
+      iex> {:error, "pouet"}
+      ...> |> ImportDataService.filter_header_csv
+      false
+
+  """
+  def filter_header_csv({:ok, %{headers: headers}}) do
+     Enum.any?(headers,
+       fn {k, v} -> k == "Content-Type" && String.contains?(v, "csv") end)
+  end
+
+  def filter_header_csv(_), do: false
+
+  defp download_csv(%{"url" => url}) do
+    case HTTPoison.get(url) do
+      {:ok, response = %{status_code: 200}} ->
+        {:ok, response}
+      {:ok, _response} ->
+        {:error, "bad status code"}
+      {:error, error} ->
+        {:error, error}
+    end
+  end
+
+  defp download_csv(resources) when is_list(resources) do
+    resources
+    |> Enum.map(&download_csv/1)
+    |> Enum.filter(&filter_header_csv/1)
+    |> case do
+      bodys = [_ | _] -> {:ok, Enum.map(bodys, fn {_, v} -> v.body end)}
+      [] -> {:error, "no csv found"}
+    end
+  end
+
+  defp download_csv(_resources) do
+    {:error, "no csv found"}
+  end
+
+  @doc """
+  Get a download from a CSVs if it exists
+
+  ## Examples
+      iex> ["name,file\\ntoulouse,http", "stop,lon,lat\\n1,48.8,2.3"]
+      ...> |> ImportDataService.get_url_from_csv()
+      "http"
+
+      iex> 
+      ...> |> ImportDataService.get_url_from_csv()
+      {:error, "no column file"}
+
+  """
+  def get_url_from_csv(bodys) when is_list(bodys) do
+    bodys
+    |> Enum.map(&get_url_from_csv/1)
+    |> Enum.filter(fn {status, _} -> status == :ok end)
+    |> case do
+      urls = [_ | _] -> {:ok, Enum.map(urls, fn {_, v} -> v end)}
+      [] -> {:error, "no url found"}
+    end
+  end
+
+  @doc """
+  Get a download from a CSV if it exists
+
+  ## Examples
+      iex> "name,file\\ntoulouse,http"
+      ...> |> ImportDataService.get_url_from_csv()
+      {:ok, "http"}
+
+      iex> "stop,lon,lat\\n1,48.8,2.3"
+      ...> |> ImportDataService.get_url_from_csv()
+      {:error, "no column file"}
+
+      iex> "Donnees;format;Download\\r\\nHoraires des lignes TER;GTFS;https\\r\\n"
+      ...> |> ImportDataService.get_url_from_csv()
+      {:ok, "https"}
+
+  """
+  def get_url_from_csv(body) do
+    _separators = [?;, ?,]
```

Comment:
No need to name this one, or if you still want to do so, add it as a @module value or something like that.